core/commands/generate_product_repositories_command.rb

```diff
@@ -224,10 +220,69 @@ def parse_maxscale_ci_deb_repository(config)
     )
   end
 
+  def get_maxscale_repository_name_for_docker(base_url, username, password)
+    uri_with_names_repos = base_url + '/v2/_catalog'
+    begin
+      doc = Nokogiri::HTML(open(uri_with_names_repos, http_basic_authentication: [username, password]))
```

Comment:
Why use Nokogiri::HTML parser first? The server provides the JSON, no need to double convert it to HTML representation first, then into the JSON.